File: src/test/java/com/weeth/domain/board/test/fixture/NoticeTestFixture.java
package com.weeth.domain.board.test.fixture;

import com.weeth.domain.board.domain.entity.Notice;
import com.weeth.domain.user.domain.entity.User;

import java.util.ArrayList;

public class NoticeTestFixture {
    public static Notice createNotice(String title, User user){
        return Notice.builder()
                .title(title)
                .content("내용")
                .user(user)
                .comments(new ArrayList<>())
                .commentCount(0)
                .build();
    }

    public static Notice createNotice(Long id, String title){
        return Notice.builder()
                .id(id)
                .title(title)
                .content("내용")
                .comments(new ArrayList<>())
                .commentCount(0)
                .build();
    }

    public static Notice createNotice(Long id, String title, User user){
        return Notice.builder()
                .id(id)
                .title(title)
                .content("내용")
                .user(user)
                .comments(new ArrayList<>())
                .commentCount(0)
                .build();
    }

    public static Notice createNotice(String title){
        return Notice.builder()
                .title(title)
                .content("내용")
                .comments(new ArrayList<>())
                .commentCount(0)
                .build();
    }
}

File: src/test/kotlin/com/weeth/domain/board/fixture/NoticeTestFixture.kt
package com.weeth.domain.board.fixture

import com.weeth.domain.board.domain.entity.Notice
import com.weeth.domain.user.domain.entity.User

object NoticeTestFixture {
    fun createNotice(
        id: Long? = null,
        title: String,
        user: User? = null,
    ): Notice =
        Notice
            .builder()
            .id(id)
            .title(title)
            .content("내용")
            .user(user)
            .comments(ArrayList())
            .commentCount(0)
            .build()
}
